### PR TITLE
chore(docker-compose): remove what is not needed

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Problem
Delete some unnecessary Docker compose related files.

## Changes

* `docker-compose.arm64.yml` was necessary as an ARM container image for ClickHouse wasn't available upstream. This is now not needed anymore since CH 22.3 includes the arch support. A PR to update our Handbook is available at: https://github.com/PostHog/posthog.com/pull/3475
* `docker-compose.test.yml` might be used somewhere but I can't find the reference. Let's see if CI breaks
* `docker-compose.yml` same as above 
* `docker-compose-config.*` those were utilities used to generate a `docker-compose.yml` based on user inputs. They are now superseeded by hobby + K8s deployment options 

The diff between `docker-compose.dev.yml`  and `docker-compose.hobby.yml` is now pretty small (try: `diff docker-compose.*`). 

## How did you test this code?

**Local development**
Followed the local development guide https://github.com/PostHog/posthog.com/pull/3475, all ✅ 

**Hobby deployment**
Installed a new hobby deployment stack using this branch on DO:
```
root@guido-delete-me:~# /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/posthog/posthog/docker_compose/bin/deploy-hobby)"
[....]
We will need to wait ~5-10 minutes for things to settle down, migrations to finish, and TLS certs to be issued

⏳ Waiting for PostHog web to boot (this will take a few minutes)
⌛️ PostHog looks up!

🎉🎉🎉  Done! 🎉🎉🎉

To stop the stack run 'docker-compose stop'
To start the stack again run 'docker-compose start'
If you have any issues at all delete everything in this directory and run the curl command again

To upgrade: run /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/posthog/posthog/HEAD/bin/upgrade-hobby)"

PostHog will be up at the location you provided!
https://test.posthog.net

It's dangerous to go alone! Take this: 🦔
```

### CODE REVIEW

**Comment by Twixes** [comment]
Isn't `docker-compose.yml` used in `deploy-hobby`?

**Comment by guidoiaquinti** [comment]
> Isn't docker-compose.yml used in deploy-hobby?

It shouldn't (see [here](https://github.com/PostHog/posthog/blob/master/bin/deploy-hobby#L146-L153)). Did I miss anything?

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -1,8 +1,15 @@
+#
+# docker-compose file used ONLY for local development.
```
Comment (Twixes) [inline docker-compose.dev.yml:2]: We could link to https://posthog.com/handbook/engineering/developing-locally#1-spin-up-external-services here for some more guidance

**Review by Twixes — APPROVED** [review]
I was tagged for review, but it's still marked as draft, so not sure. In any case, +1 from me.

Just one extra bit that needs to be done along with this is updating our "Developing locally" guide so that people don't try to run `docker-compose.arm64.yml`: https://posthog.com/handbook/engineering/developing-locally#1-spin-up-external-services
Can you take care of it?

**Comment by guidoiaquinti** [comment]
> I was tagged for review, but it's still marked as draft, so not sure. In any case, +1 from me.

Yep, sorry I was going to mark it ready for review tomorrow after completing some tests

> Just one extra bit that needs to be done along with this is updating our "Developing locally" guide so that people don't try to run `docker-compose.arm64.yml`: https://posthog.com/handbook/engineering/developing-locally#1-spin-up-external-services Can you take care of it?

Already done (see PR description above).

Thank you for your review!

**Review by hazzadous — APPROVED** [review]

**Pull request merged**